ittapi-rs/ittapi/src/task.rs

```diff
@@ -0,0 +1,72 @@
+use crate::{domain::Domain, string::StringHandle};
+
+/// A task is a logical unit of work performed by a particular thread. See the [Task API]
+/// documentation for more information.
+///
+/// [Task API]:
+///     https://www.intel.com/content/www/us/en/develop/documentation/vtune-help/top/api-support/instrumentation-and-tracing-technology-apis/instrumentation-tracing-technology-api-reference/task-api.html
+///
+/// ```
+/// # use ittapi::{Domain, StringHandle, Task};
+/// let _env_path = scoped_env::ScopedEnv::set("INTEL_LIBITTNOTIFY64", "<some path>");
+/// let domain = Domain::new("domain");
+/// let name = StringHandle::new("task");
+/// // Measure a task using a pre-initialized string handle (most efficient).
+/// {
+///     let task = Task::begin(&domain, name);
+///     let _ = 2 + 2;
+///     task.end(); // Task ended here.
+///     let _ = 3 + 3;
+/// }
+/// // Measure a task using a string reference (most convenient).
+/// {
+///     let _task = Task::begin(&domain, "task");
+///     let _ = 2 + 2;
+///     // Task ended here, when dropped.
+/// }
+/// ```
+pub struct Task<'a>(&'a Domain);
+impl<'a> Task<'a> {
+    /// Start a task on the current thread.
+    pub fn begin(domain: &'a Domain, name: impl Into<StringHandle>) -> Self {
+        if let Some(create_fn) = unsafe { ittapi_sys::__itt_task_begin_ptr__3_0 } {
+            // Currently the `taskid` and `parentid` parameters are unimplemented (TODO).
+            unsafe { create_fn(domain.as_ptr(), ITT_NULL, ITT_NULL, name.into().as_ptr()) };
+        }
```

Comment:
I'm agree that ITT API should just work and do nothing in case of actual collector is absent. We do not need to notify customer about dynamic library's status (absence/present) automatically - only in case of direct call to "__itt_get_collection_state" API.